__tests__/html/hooks.useScrollUpDown.html

```diff
@@ -57,12 +57,12 @@
         expect(scrollable.scrollTop).toBe(scrollTopAtBottommost);
 
         // Call useScrollUp() should scroll the transcript view up
-        await renderWithFunction(() => useScrollUp()());
+        await renderWithFunction(() => Promise.resolve(useScrollUp()).then(fn => fn()));
```

Comment:
Curious why we need to `Promise.resolve()` here.

Reply:
As we use `usePropagate` we can't call the callback in render. I can rework to `setImmediate` instead. I just didn't figure out how I'd use it in the context. It could be:
```
        await renderWithFunction(() => setImmediate(fn =>fn(), useScrollUp());
```
And it is [not recommended for usage](https://developer.mozilla.org/en-US/docs/Web/API/Window/setImmediate)

Reply:
Oh I see, you are right.

There is an escape hatch there... `allowPropagateDuringRender: true`. I think we could use it here.

tl;dr "if you promise not to re-render, I am okay that you call `usePropagate`."

Reply:
My bad... we shouldn't allow propagate in almost/all circumstances.